src/common/helpers/snap-builds.js

```diff
@@ -0,0 +1,91 @@
+//   architecture: 'i386', // last part of 'distro_arch_series_link'
+//
+//   // TODO: full list of statuses based on
+//   // https://git.launchpad.net/launchpad/tree/lib/lp/buildmaster/enums.py#n22
+//   status:  'error', // parsed based on 'buildstate' -> success, error, pending
+//   statusMessage: 'Failed to build', // 'buildstate'
+//
+//   dateCreated: '2016-12-01T17:08:36.317805+00:00', // 'datecreated'
+//   dateStarted: '2016-12-01T17:08:36.317805+00:00', // 'date_started'
+//   dateFirstDispatched: '2016-12-01T17:08:36.317805+00:00', // 'date_first_dispatched'
+//   dateCompleted: '2016-12-01T17:10:36.317805+00:00', // 'datebuilt'
+//   duration: '0:02:00.124039' // 'duration'
+// };
+
+// Based on BuildStatus from LP API
+// https://git.launchpad.net/launchpad/tree/lib/lp/buildmaster/enums.py#n22
+export const BuildStatusLP = {
+  NEEDSBUILD: 'Needs building',     // pending
+  FULLYBUILT: 'Successfully built', // success
+  FAILEDTOBUILD:'Failed to build',  // error
+  MANUALDEPWAIT: 'Dependency wait', // pending
+  CHROOTWAIT: 'Chroot problem',     // error
+  SUPERSEDED: 'Build for superseded Source', // error
+  BUILDING: 'Currently building',   // pending
+  FAILEDTOUPLOAD: 'Failed to upload', // error
+  UPLOADING: 'Uploading build',     // pending
+  CANCELLING: 'Cancelling build',   // pending
+  CANCELLED: 'Cancelled build'      // error
+};
+
+export const BuildStatus = {
+  SUCCESS: 'success', // for builds successfully finished
+  PENDING: 'pending', // for build currently running or in any way in progress
+  ERROR: 'error'      // for builds failed for any reason
+};
+
+// TODO:
+// export enum with build status success/pending/error
+// or export full list of statuses (like in API) and let view do the mapping?
+
+// mapping between build status from LP and pending/success/error internal status
+const BuildStatusMapping = {
+  [BuildStatusLP.NEEDSBUILD]: BuildStatus.PENDING,
+  [BuildStatusLP.FULLYBUILT]: BuildStatus.SUCCESS,
+  [BuildStatusLP.FAILEDTOBUILD]: BuildStatus.ERROR,
+  [BuildStatusLP.MANUALDEPWAIT]: BuildStatus.PENDING,
+  [BuildStatusLP.CHROOTWAIT]: BuildStatus.ERROR,
+  [BuildStatusLP.SUPERSEDED]: BuildStatus.ERROR,
+  [BuildStatusLP.BUILDING]: BuildStatus.PENDING,
+  [BuildStatusLP.FAILEDTOUPLOAD]: BuildStatus.ERROR,
+  [BuildStatusLP.UPLOADING]: BuildStatus.PENDING,
+  [BuildStatusLP.CANCELLING]: BuildStatus.PENDING,
+  [BuildStatusLP.CANCELLED]: BuildStatus.ERROR
+};
+
+function getLastPartOfUrl(url) {
+  return url ? url.substr(url.lastIndexOf('/') + 1) : null;
+}
+
+export function snapBuildFromAPI(entry) {
+  return entry ? {
+    buildId: getLastPartOfUrl(entry.self_link),
+    buildLogUrl: entry.build_log_url,
+
+    architecture: getLastPartOfUrl(entry.distro_arch_series_link),
+
+    status: BuildStatusMapping[entry.buildstate],
+    statusMessage: entry.buildstate,
+
+    dateCreated: entry.datecreated,
+    dateStarted: entry.date_started,
+    dateFirstDispatched: entry.date_first_dispatched,
```

Comment:
You don't use `dateFirstDispatched`, and I think every possible use of it on this site should probably use `dateStarted` instead, so I'd drop this.

I think `dateCreated` is currently unused too, but I can imagine reasonable uses of that here.